### PR TITLE
Cleanup unused dirs from `store/` on all disks

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Follow-up to #38265

### CODE REVIEW

File: src/Interpreters/DatabaseCatalog.cpp [hunk 1]
```diff
@@ -894,7 +891,7 @@ void DatabaseCatalog::enqueueDroppedTableCleanup(StorageID table_id, StoragePtr
             create->setTable(table_id.table_name);
             try
             {
-                table = createTableFromAST(*create, table_id.getDatabaseName(), data_path, getContext(), false).second;
```
Comment (antonio2368) [inline src/Interpreters/DatabaseCatalog.cpp:897]: why was this changed?

**Review by antonio2368 — APPROVED** [review]

File: src/Interpreters/DatabaseCatalog.cpp (hunk 1 above)
Comment (tavplubix) [reply src/Interpreters/DatabaseCatalog.cpp:897]: It's not related to `store/` cleanup, but it's better to set `force_restore` to `true` here to avoid unnecessary exceptions, because we don't really care about data when we load it just to drop.

**Comment by tavplubix** [comment]
Stateless tests (ubsan) - https://github.com/ClickHouse/ClickHouse/pull/39720#issuecomment-1205573182
Stress test (undefined) - #39857
